api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -572,64 +572,32 @@ public RelationshipType saveRelationshipType(RelationshipType relationshipType)
 		
 		String attrString = "";
 		
-		final String fatalString = "Should not be here.";
-		
 		// TODO cache the global properties to speed this up??
 		// Is hibernate taking care of caching and not hitting the db every time? (hopefully it is)
 		if (viewType == null) {
 			return Context.getPersonService().getAllPersonAttributeTypes();
 		} else if (viewType == ATTR_VIEW_TYPE.LISTING) {
 			String patientListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_LISTING_ATTRIBUTES, "");
 			String userListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_LISTING_ATTRIBUTES, "");
-			if (personType == null || personType == PERSON_TYPE.PERSON) {
-				attrString = patientListing + "," + userListing;
-			} else if (personType == PERSON_TYPE.PATIENT) {
-				attrString = patientListing;
-			} else if (personType == PERSON_TYPE.USER) {
-				attrString = userListing;
-			} else {
-				log.error(MarkerFactory.getMarker("FATAL"), fatalString);
-			}
+			attrString = combineAttributes(userListing, patientListing, personType);
```

Comment:
@Ankitkumar94 you need to switch the arguments when calling the method, check all occurences. you call the method with user then patient attributes but the method expects them in different order. please switch them here.